src/processor/mod.rs

```diff
@@ -200,7 +248,7 @@ impl<'a> TransactionProcessor<'a> {
             }
```

Comment:
unregister should be checked before the above if condition. Otherwise a unregister request would be attempted on a closed channel which is not pleasant.